dvc/repo/add.py

```diff
@@ -255,12 +255,9 @@ def _create_stages(
             external=external,
         )
         restore_meta(stage)
-        if stage.can_be_skipped:
-            stage = None
```

Comment:
hmm, I find this helpful though.

Reply:
@skshetry But it was only used for this semi-run-cache, while it wasn't the behavior in real run-cache. So inconsistent and confusing :slightly_frowning_face: Thinking if maybe we could introduce some flag that will signal that the stage was actually ran. E.g. `if stage.ran: ...` or something.

Reply:
Or `stage.restored`. Seems to be more useful than returning `None`, as you have more information and more control.

Reply:
but, for `dvc add`-ed files, it would use `changed_outs()` result to determine this, no?

Reply:
@skshetry You mean `ran/restored`? Those are close to meaningless for `dvc add`, as they are more of a pipeline thing. We were calling `commit` in `is_cached` anyway, so without it we'll just verify that outputs and cache are in place. If anyone needs to know whether or not there have been any workspace changes, they could call `status` before running `dvc add`.

Reply:
The `stage = None` is definitely confusing, there's even a comment for that:

https://github.com/iterative/dvc/blob/9ad1e0e15925ec3ca0d741d64bdf65d293a83f58/dvc/repo/trie.py#L10

But I find this hack simpler than proper solution. The `dvc add` has been kind of broken for `--file` for long, but this might break for non `--file` case as well, by removing `.dvc` file.

Consider the following scenario:
```console
$ echo foo > foo
$ echo bar > bar
$ dvc add foo bar
$ ls
bar bar.dvc foo foo.dvc 
$ rm foo
$ echo foobar > bar
$ dvc add foo bar
$ ls  # no foo.dvc file
bar bar.dvc
```
```

Reply:
Oh, good point! Need to remove that filtering too.

Reply:
I meant the following:
```console
$ dvc add foo
Stage is cached, skipping                                                                        
  0% Add|                                                               |0/0 [00:00,     ?file/s]
```
vs the current PR:
```console
$ dvc add foo
100% Add|███████████████████████████████████████████████████████████████|1/1 [00:00,  4.34file/s]

To track the changes with git, run:

	git add foo.dvc
```

Reply:
@skshetry Previous one was confusing https://github.com/iterative/dvc/issues/4529 .

The current one is related to the way we have the progress bar in `add()`. Definitely could consider running the targets through `dvc status` first to filter-out unchanged ones and not reflect them in the progress bar. Or to somehow pass the pbar into the saving logic and let it handle the progress. To me 0% seem confusing, and the current PR behavior makes more sense, as it is a corner case of you already having the data in the cache (e.g. some collegue already added it to the sshared cache, or you've already added it in some other branch).

Reply:
My concern was with "is cached" part, which I find helpful, and there's no need for a hint in that case as well. But, fine for now. 
I'll create a separate issue with `dvcfile.remove()` which I find unsafe here.

Reply:
That issue already exists 😄  https://github.com/iterative/dvc/issues/3362